### ISSUE TITLE
geoserver - icu4j does not recognize Java patch version > 255

### ISSUE THREAD

**Issue by pmauduit** [georchestra/georchestra#3107]
On too recent Java8 runtimes (where the patch number in the version is > 255), icu4j throws an exception considering the java version as invalid, which makes geoserver unable to boot, see reported upstream issue:

https://osgeo-org.atlassian.net/browse/GEOS-9702

Affected java version:
```
% java -version
openjdk version "1.8.0_262"
OpenJDK Runtime Environment (build 1.8.0_262-b10)
OpenJDK 64-Bit Server VM (build 25.262-b10, mixed mode)
```
Here is the last part of the stacktrace encountered at GS boot:

```
Caused by: 
java.lang.NoClassDefFoundError: Could not initialize class com.ibm.icu.impl.ICUDebug
	at com.ibm.icu.impl.NormalizerDataReader.<clinit>(NormalizerDataReader.java:300)
```

Encountered on a GS 2.17 from geOrchestra (see related branch), but should also affect previous version, which should make use of the same icu4j jar (3.4.4).

**Comment by pmauduit** [comment]
> facepalm

checking more thoroughly (see issue on osgeo's JIRA), it seems that the icu4j is required by the arcsde plugin (which is required in case of activating the gs-importer plugin), and should not be needed in a "vanilla" GS:

```
[INFO] +- org.geoserver.importer:gs-importer-core:jar:2.17.0:compile
[INFO] |  +- org.geotools.jdbc:gt-jdbc-h2:jar:23.0:compile
[INFO] |  |  \- org.opengeo:geodb:jar:0.9:compile
[INFO] |  |     \- net.sourceforge.hatbox:hatbox:jar:1.0.b10:compile
[INFO] |  +- org.geotools.jdbc:gt-jdbc-oracle:jar:23.0:compile
[INFO] |  +- org.geotools.jdbc:gt-jdbc-sqlserver:jar:23.0:compile
[INFO] |  |  +- net.sourceforge.jtds:jtds:jar:1.3.1:compile
[INFO] |  |  \- com.microsoft.sqlserver:mssql-jdbc:jar:6.1.0.jre7:compile
[INFO] |  |     \- com.microsoft.azure:azure-keyvault:jar:0.9.3:compile
[INFO] |  |        +- com.microsoft.azure:azure-core:jar:0.9.3:compile
[INFO] |  |        |  +- com.sun.jersey:jersey-client:jar:1.13:compile
[INFO] |  |        |  |  \- com.sun.jersey:jersey-core:jar:1.13:compile
[INFO] |  |        |  \- com.sun.jersey:jersey-json:jar:1.13:compile
[INFO] |  |        |     +- com.sun.xml.bind:jaxb-impl:jar:2.2.3-1:compile
[INFO] |  |        |     +- org.codehaus.jackson:jackson-core-asl:jar:1.9.2:compile
[INFO] |  |        |     +- org.codehaus.jackson:jackson-mapper-asl:jar:1.9.2:compile
[INFO] |  |        |     +- org.codehaus.jackson:jackson-jaxrs:jar:1.9.2:compile
[INFO] |  |        |     \- org.codehaus.jackson:jackson-xc:jar:1.9.2:compile
[INFO] |  |        +- javax.inject:javax.inject:jar:1:compile
[INFO] |  |        \- com.microsoft.azure:adal4j:jar:1.0.0:compile
[INFO] |  |           +- com.nimbusds:oauth2-oidc-sdk:jar:4.5:compile
[INFO] |  |           |  +- net.jcip:jcip-annotations:jar:1.0:compile
[INFO] |  |           |  +- net.minidev:json-smart:jar:1.1.1:compile
[INFO] |  |           |  +- com.nimbusds:lang-tag:jar:1.4:compile
[INFO] |  |           |  \- com.nimbusds:nimbus-jose-jwt:jar:3.1.2:compile
[INFO] |  |           \- com.google.code.gson:gson:jar:2.8.5:compile
[INFO] |  +- org.geotools:gt-arcsde:jar:23.0:compile
[INFO] |  |  +- org.geotools:gt-arcsde-common:jar:23.0:compile
[INFO] |  |  |  \- com.ibm.icu:icu4j:jar:3.4.4:compile
[INFO] |  |  \- jsqlparser:jsqlparser:jar:0.3.14:compile
```

If one is using such a plugin, we shall see if it is safe to exclude icu4j from the webapp classpath and/or consider removing the arcsde plugin from the dependencies.

**Comment by landryb** [comment]
Fwiw, hitting this issue too on buster with recent adoptopenjdk release, trying to deploy 20.0.5/gs 2.16. What should be done to really fix it ? Removing arcsde plugin from the build ? Dropping the icu4j jar from the webapp works around the startup issue here.

**Comment by fvanderbiest** [comment]
> What should be done to really fix it ?

All of the above, yes. 
Either removing arcsde plugin or excluding the icu4j jar.

**Comment by landryb** [comment]
from looking at the geoserver issue; the dependency from the importer on the arcsde bit was removed, but unfortunately the commit isnt linked to it

**Comment by landryb** [comment]
ah, https://github.com/geoserver/geoserver/commit/34b88a34de33ff49e26b29fe8924f18ddd24bebc is a winner

**Comment by landryb** [comment]
and backported upstream to 2.16.x branch in https://github.com/geoserver/geoserver/pull/4435

**Comment by pmauduit** [comment]
I wonder if it is correctly fixed with the previous PR actually. Relying on the official JAR in previous versions that have been released, e.g.:
https://mvnrepository.com/artifact/org.geoserver.importer/gs-importer-core/2.17.0

there is still a dependency to gt-arcsde. If we want a custom version of the gs-importer-core plugin, we would have to do the same as the other georchestra-customized modules, like:
https://github.com/georchestra/georchestra/blob/master/geoserver/pom.xml#L30-L33

**Comment by landryb** [comment]
or wait for geoserver 2.17.3 which will include geoserver/geoserver#4434 ? :)

**Comment by pmauduit** [comment]
Already fixed with 2.17.2 & 
https://github.com/georchestra/georchestra/commit/c2ed6682db9f2a51931ebebfca7ed11ebd81ae80
